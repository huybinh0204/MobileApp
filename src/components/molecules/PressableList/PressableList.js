import PropTypes from 'prop-types';
import React, { memo } from 'react';
import { FlatList } from 'react-native';
import { PressableListItem } from '_components/atoms/index';
import { Separator } from './PressableList.styles';

const PressableList = ({ items, ...rest }) => (
  <FlatList
    {...rest}
    data={items}
    ItemSeparatorComponent={({ leadingItem }) => (leadingItem.hide ? null : <Separator />)}
    keyExtractor={(item) => item.title}
    renderItem={({ item }) => {
      return item.hide ? null : (
        <PressableListItem
          title={item.title}
          leftItem={item.leftItem}
          rightItem={item.rightItem}
          onPress={item.onPress}
          testID={item.testID}
        />
      );
    }}
  />
);

PressableList.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      hide: PropTypes.bool,
      title: PropTypes.string.isRequired,
      leftItem: PropTypes.func,
      rightItem: PropTypes.func,
      onPress: PropTypes.func,
    })
  ).isRequired,
};

export default memo(PressableList);
