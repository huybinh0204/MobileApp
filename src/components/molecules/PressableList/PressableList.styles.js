import styled from '@emotion/native';

export const Separator = styled.View(({ theme }) => ({
  width: '100%',
  height: 1,
  borderColor: theme.colors.beta50,
  borderWidth: 1,
}));
