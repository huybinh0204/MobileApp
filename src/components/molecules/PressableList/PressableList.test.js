import { render } from '@testing-library/react-native';
import React from 'react';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import PressableList from './PressableList';

describe('PressableList', () => {
  let props;

  beforeEach(() => {
    props = {
      items: [{ title: 'List Item 1' }, { title: 'List Item 2' }],
    };
  });

  it('should match the snapshot', () => {
    const { toJSON } = render(
      <ComponentWithProviders component={PressableList} componentProps={props} />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should display correct number of items', () => {
    const { getAllByTestId } = render(
      <ComponentWithProviders component={PressableList} componentProps={props} />
    );

    expect(getAllByTestId('ItemTitle')).toHaveLength(2);
  });
});
