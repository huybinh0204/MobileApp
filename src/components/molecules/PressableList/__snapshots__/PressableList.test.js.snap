// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PressableList should match the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTScrollView
                        ItemSeparatorComponent={[Function]}
                        data={
                          Array [
                            Object {
                              "title": "List Item 1",
                            },
                            Object {
                              "title": "List Item 2",
                            },
                          ]
                        }
                        getItem={[Function]}
                        getItemCount={[Function]}
                        keyExtractor={[Function]}
                        onContentSizeChange={[Function]}
                        onLayout={[Function]}
                        onMomentumScrollBegin={[Function]}
                        onMomentumScrollEnd={[Function]}
                        onScroll={[Function]}
                        onScrollBeginDrag={[Function]}
                        onScrollEndDrag={[Function]}
                        removeClippedSubviews={false}
                        renderItem={[Function]}
                        scrollEventThrottle={50}
                        stickyHeaderIndices={Array []}
                        viewabilityConfigCallbackPairs={Array []}
                      >
                        <View>
                          <View
                            onLayout={[Function]}
                            style={null}
                          >
                            <View
                              accessible={true}
                              collapsable={false}
                              focusable={false}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "alignItems": "center",
                                  "alignSelf": "stretch",
                                  "flexDirection": "row",
                                  "minHeight": 50,
                                  "opacity": 1,
                                }
                              }
                              testID="PressableListItemContainer"
                            >
                              <Text
                                allowFontScaling={false}
                                marginLeft={false}
                                style={
                                  Array [
                                    Object {
                                      "flex": 1,
                                      "fontFamily": "Poppins-SemiBold",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                      "marginLeft": 0,
                                    },
                                    null,
                                  ]
                                }
                                testID="ItemTitle"
                              >
                                List Item 1
                              </Text>
                            </View>
                            <View
                              style={
                                Array [
                                  Object {
                                    "borderColor": "#F2F3F4",
                                    "borderWidth": 1,
                                    "height": 1,
                                    "width": "100%",
                                  },
                                  undefined,
                                ]
                              }
                            />
                          </View>
                          <View
                            onLayout={[Function]}
                            style={null}
                          >
                            <View
                              accessible={true}
                              collapsable={false}
                              focusable={false}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "alignItems": "center",
                                  "alignSelf": "stretch",
                                  "flexDirection": "row",
                                  "minHeight": 50,
                                  "opacity": 1,
                                }
                              }
                              testID="PressableListItemContainer"
                            >
                              <Text
                                allowFontScaling={false}
                                marginLeft={false}
                                style={
                                  Array [
                                    Object {
                                      "flex": 1,
                                      "fontFamily": "Poppins-SemiBold",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                      "marginLeft": 0,
                                    },
                                    null,
                                  ]
                                }
                                testID="ItemTitle"
                              >
                                List Item 2
                              </Text>
                            </View>
                          </View>
                        </View>
                      </RCTScrollView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
